R/plotGMA.R

```diff
@@ -0,0 +1,270 @@
+                 values_to = "Met_count") %>% 
+    dplyr::select(-contains("NCG")) %>% 
+    replace_na(list(Met_count = 0)) %>% 
+    left_join(x = .,
+              y = DEGs_logFC,
+              by = "Hugo_Symbol")
+  
+  ## Genelist mode
+  
+  # If genelist is supplied
+  if (length(genelist) > 0) {
+    
+    # Subset data to contain only supplied genes in genelist
+    summary_heatmap <- summary_heatmap %>% 
+      filter(Hugo_Symbol %in% genelist)
+    
+    # Get number of Moonlight_Oncogenic_Mediator groups in subsetted data
+    driver_groups <- summary_heatmap$Moonlight_Oncogenic_Mediator %>% 
+      unique %>% 
+      length
+    
+    # If genelist contain both oncogenic mediator gene types
+    if (driver_groups == 2) {
+      
+      # Create heatmap of genes in genelist 
+      # grouped by oncogenic mediator types
+      met_heatmap <- summary_heatmap %>% 
+        group_by(Moonlight_Oncogenic_Mediator) %>% 
+        heatmap(.data = .,
+                .row = Met_type,
+                .column = Hugo_Symbol, 
+                .value = Met_count,
+                cluster_rows = FALSE,
+                cluster_columns = TRUE,
+                show_column_dend = FALSE,
+                show_column_names = TRUE,
+                scale = "none",
+                palette_value = c("white", "blue", "darkblue"),
+                column_title = "Heatmap of methylation state by EpiMix") %>% 
+        add_tile(Moonlight_Oncogenic_Mediator, 
+                 palette = c("goldenrod2", "dodgerblue3")) %>% 
+        add_tile(logFC, 
+                 palette = c("chartreuse4",  "firebrick3"))
+        save_pdf(met_heatmap, 
+                 height = 15, 
+                 width = 35, 
+                 units = "cm", 
+                 filename = paste(additionalFilename,
+                                  "heatmap_genelist_met.pdf", 
+                                  sep = ""))
+      
+      # If genelist only contain one oncogenic mediator gene type
+    } else {
+      
+      # Create heatmap of genes in genelist
+      met_heatmap <- summary_heatmap %>% 
+        heatmap(.data = .,
+                .row = Met_type,
+                .column = Hugo_Symbol, 
+                .value = Met_count,
+                cluster_rows = FALSE,
+                cluster_columns = TRUE,
+                show_column_dend = FALSE,
+                show_column_names = TRUE,
+                scale = "none",
+                palette_value = c("white", "blue", "darkblue"),
+                column_title = "Heatmap of methylation state by EpiMix")  %>% 
+        add_tile(logFC, 
+                 palette = c("chartreuse4",  "firebrick3")) 
+        save_pdf(met_heatmap, 
+                 height = 15, 
+                 width = 35, 
+                 units = "cm", 
+                 filename = paste(additionalFilename,
+                                  "heatmap_genelist_met.pdf", 
+                                  sep = ""))
+      
+    }
```

Comment:
bit of repeated code in this section - can we maybe do something like

```
met_heatmap <- summary_heatmap
if (driver_groups == 2) {
    met_heatmap <- met_heatmap %>% group_by(Moonlight_Oncogenic_Mediator)
}

met_heatmap %>% heatmap(...
```

Reply:
updated